miner/src/miner.rs

```diff
@@ -88,25 +88,24 @@ impl Miner {
 
     fn check_seal(&mut self, pow_hash: H256, seal: Seal) {
         if let Some(work) = self.works.lock().get_refresh(&pow_hash) {
-            let block = &work.block;
             if self
                 .pow
-                .verify_proof_difficulty(&seal.proof(), &block.header().difficulty())
+                .verify_proof_difficulty(&seal.proof(), &work.block.header().difficulty())
             {
                 self.notify_workers(WorkerMessage::Stop);
+                let raw_header = work.block.header().raw().to_owned();
+                let block = BlockBuilder::from_block(work.block.clone())
+                    .header(raw_header.with_seal(seal))
+                    .build();
+
                 debug!(
-                    "found seal {:?} of block #{} {:x}",
-                    seal,
+                    "Found! #{} {:x}",
```

Comment:
Use `{:#x}` to print hash. So does the line 118

Reply:
Resolved